src/Schema/Column/BigIntColumn.php

```diff
@@ -55,4 +61,13 @@ public function phpTypecast(mixed $value): string|null
 
         return (string) $value;
     }
+
+    protected function dbTypecastString(string $value): int|string|null
+    {
+        if ($value === '') {
+            return null;
+        }
+
+        return PHP_INT_MAX >= $value && $value >= PHP_INT_MIN ? (int) $value : $value;
```

Comment:
That was always quite inconvenient/dangerous part. You write the code for `int` and then for some numbers it returns `string` and everything breaks. I know that return type suggests that a string could be there but still that was the case for Yii2.

Reply:
Not sure if there's anything we can do about it though. So not a blocker to merge this PR.

Reply:
This is typecasting for DB, some queries are faster if pass `int` instead of `string` where `int` is expected.
When typecasting for PHP it is always `string` or `null`